SessionPlayer/HUD.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import { View, Text, StyleSheet } from 'react-native';
+
+export default class HUD extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            timer: 0,
+        };
+
+        this.timerIntervalID = setInterval(() => {
+            this.setState({
+                timer: this.state.timer + 1
+            });
+        }, 1000);
+    }
+
+    componentWillUnmount() {
+        clearInterval(this.timerIntervalID);
+    }
+
+    componentDidUpdate() {
+        const { guideState } = this.props;
+
+        if (
+            this.timerIntervalID
+            && (guideState === 'finished' || guideState === 'all_finished')
```

Comment:
I think `all_finished` is an ambiguous state. I believe `guide_finished` makes a little more sense (or something similar).

Reply:
No, I think it needs to be `session_finished`.